src/backend_model_instance.cc

```diff
@@ -193,135 +186,62 @@ TritonModelInstance::~TritonModelInstance()
-        RETURN_IF_ERROR(ResetNumaMemoryPolicy());
-        RETURN_IF_ERROR(err);
-        RETURN_IF_ERROR(
-            model->RegisterInstance(std::move(new_instance), passive));
-
-        // When deploying on GPU, we want to make sure the GPU memory usage
-        // is within allowed range, otherwise, stop the creation to ensure
-        // there is sufficient GPU memory for other use.
-        // We check the usage after loading the instance to better enforcing
-        // the limit. If we check before loading, we may create instance
-        // that occupies the rest of available memory which against the purpose
-        if (kind == TRITONSERVER_INSTANCEGROUPKIND_GPU) {
-          size_t free, total;
-          double memory_limit;
-          RETURN_IF_ERROR(GetDeviceMemoryInfo(id, &free, &total));
-          RETURN_IF_ERROR(BackendConfigurationModelLoadGpuFraction(
-              backend_cmdline_config_map, id, &memory_limit));
-          const size_t allow = total * memory_limit;
-          const size_t used = total - free;
-          if (used > allow) {
-            return Status(
-                Status::Code::UNAVAILABLE,
-                std::string("can not create model '") + instance_name +
-                    "': memory limit set for " +
-                    TRITONSERVER_InstanceGroupKindString(kind) + " " +
-                    std::to_string(id) +
-                    " has exceeded, model loading is rejected.");
-          }
-        }
-      }
+  RETURN_IF_ERROR(SetNumaConfigOnThread(*host_policy));
+  auto err = ConstructAndInitializeInstance(
+      model, name, signature, kind, device_id, profile_names, passive,
+      host_policy_name, *host_policy, rate_limiter_config, secondary_devices,
+      triton_model_instance);
+  RETURN_IF_ERROR(ResetNumaMemoryPolicy());
+  RETURN_IF_ERROR(err);
+
+  // When deploying on GPU, we want to make sure the GPU memory usage
+  // is within allowed range, otherwise, stop the creation to ensure
+  // there is sufficient GPU memory for other use.
+  // We check the usage after loading the instance to better enforcing
+  // the limit. If we check before loading, we may create instance
+  // that occupies the rest of available memory which against the purpose
+  if (kind == TRITONSERVER_INSTANCEGROUPKIND_GPU) {
+    size_t free, total;
+    double memory_limit;
+    RETURN_IF_ERROR(GetDeviceMemoryInfo(device_id, &free, &total));
+    RETURN_IF_ERROR(BackendConfigurationModelLoadGpuFraction(
+        model->BackendConfigMap(), device_id, &memory_limit));
+    const size_t allow = total * memory_limit;
+    const size_t used = total - free;
+    if (used > allow) {
+      return Status(
+          Status::Code::UNAVAILABLE,
+          std::string("can not create model '") + name +
+              "': memory limit set for " +
+              TRITONSERVER_InstanceGroupKindString(kind) + " " +
+              std::to_string(device_id) +
+              " has exceeded, model loading is rejected.");
```

Comment:
Can you move this to a helper?
```cpp
  if (kind == TRITONSERVER_INSTANCEGROUPKIND_GPU) {
    RETURN_IF_ERROR(VerifyModelLoadGpuFraction(...));
  }

  return Status::Success;
```

Reply:
Sure. Updated on commit: [Move verify model load gpu fraction to helper function](https://github.com/triton-inference-server/core/pull/207/commits/a05d4f9378bc2d8f7f69cd77c0838146f94a9e2d)